transport/uml/scp/scp_hld.puml

```diff
@@ -0,0 +1,143 @@
+rnote over MessagingClient
+  Given:
+  * Session is configured.
+end rnote
+Shyrka --> MessagingClient : SessionResponse
+MessagingClient -> MessagingClient : extract SCP from SessionResponse \n(allowed fileTypes, blockedExtensions, maxFileSize)
+MessagingClient -> MessagingClient : store supportedContentProfile as field and expose it via API
+MessagingClient --> App : State.Configured
+User -> App : select file to upload
+App -> MessagingClient : attach(byteArray, filename, uploadListener)
+MessagingClient -> MessagingClient : generateAttachmentId
+MessagingClient -> AttachmentHandler : prepare(attachmentId, byteArray, filename, uploadListener)
+AttachmentHandler -> AttachmentHandler : validateAttachment
+alt #$Error fileType != allowed || fileType == blockedExtensions || fileSize > maxFileSize
+AttachmentHandler -> MessagingClient : onError
+MessagingClient -> App : attach() throws InvalidArgumentException()
+rnote over MessagingClient
+Stop attachment process
+end rnote
+end
+AttachmentHandler -> MessageStore : updateAttachmentStateWith(attachmentId, State.Presigning)
+MessageStore -> MessageStore : add attachment to pendingMessage
+MessageStore -> App : messageListener(MessageEvent(AttachmentUpdated(attachmentId, State.Presigning))
+AttachmentHandler --> MessagingClient : return OnAttachmentRequest
+MessagingClient -> Shyrka : send OnAttachmentRequest
+MessagingClient --> App : attach() returns attachmentId
+Shyrka -> GenesysCloud : getPresignedUrl
+GenesysCloud -> S3 : getPresignedUrl
+S3 --> GenesysCloud : return PresignedUrl
+GenesysCloud --> Shyrka : return PresignedUrl
+alt #$Error GenerateUrlError. BE failed to validate file or generate upload url
+S3 --> GenesysCloud : GeneratePresignedUrlError
+GenesysCloud --> Shyrka : GeneratePresignedUrlError
+Shyrka --> MessagingClient : onMessage(GenerateUrlError(attachmentId, errorCode, errorMessage))
+MessagingClient -> AttachmentHandler : onError(UploadFailureEvent(attachmentId, errorCode, errorMessage))
+AttachmentHandler -> AttachmentHandler : processedAttachments.remove(attachmentId)
+AttachmentHandler -> MessageStore : updateAttachmentStateWith(attachmentId, State.Error(errorCode, errorMessage)
+MessageStore -> MessageStore : remove attachment from pendingMessage
+MessageStore -> App : messageListener(MessageEvent(AttachmentUpdated(attachmentId, State.Error(errorCode, errorMessage)))
+rnote over MessagingClient
+Stop attachment process
+end rnote
+end
+Shyrka --> MessagingClient : onMessage(PresignedUrlResponse(id,headers,uploadUrl))
+MessagingClient -> AttachmentHandler : upload(PresignedUrlResponse)
+AttachmentHandler -> AttachmentHandler : findProcessedAttachmentById(PresignedUrlResponse.attachmentId)
+AttachmentHandler -> MessageStore : updateAttachmentStateWith(attachmentId, State.Uploading)
+MessageStore -> MessageStore : update attachment state in pendingMessage
+MessageStore -> App : messageListener(MessageEvent(AttachmentUpdated(attachmentId, State.Uploading))
```

Comment:
Since `MessageStore` is an internal class, can we report the status to `MessagingClient`, which can then pass it to the app?